onnxruntime/core/providers/migraphx/migraphx_execution_provider_utils.h

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+
+#pragma once
+#include "core/session/onnxruntime_cxx_api.h"
+#include "core/framework/allocatormgr.h"
+#include "core/framework/execution_provider.h"
+
+namespace onnxruntime {
+
+bool IsGraphInput(const GraphViewer& graph, const std::string& name)
```

Comment:
>GraphViewer [](http://example.com/codeflow?start=24&length=11)

I wonder you can pass graph object throughout the code, and leverage graph_utils.cc functionalities.
If you prefer graph_viewer object, you can search the code for example: 
GraphViewer graph_viewer(graph);

The basic tests might have been covered by graph_test.cc, thus you might need most of them. Perhaps some UTs related to the bug you are fixing would be helpful.

Reply:
Actually, I was using those utility function previously. But after some api change, it looks like I cannot call them anymore, then I switched to the current implementation.

Reply:
Ah, I see that in the history. What's the error you are getting? Is it due to ORT code refactoring on TensorRT EP?

Reply:
> Ah, I see that in the history. What's the error you are getting? Is it due to ORT code refactoring?

I guess you mean this? https://github.com/microsoft/onnxruntime/pull/9272